pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -114,12 +116,21 @@ func (s *NodeController) Run(period time.Duration, syncNodeList bool) {
 		}, period)
 	}
 
-	// Start syncing node status.
-	go util.Forever(func() {
-		if err = s.SyncNodeStatus(); err != nil {
-			glog.Errorf("Error syncing status: %v", err)
-		}
-	}, period)
+	if *syncNodeStatus {
+		// Start syncing node status.
+		go util.Forever(func() {
+			if err = s.SyncNodeStatus(); err != nil {
+				glog.Errorf("Error syncing status: %v", err)
+			}
+		}, period)
+	} else {
+		// Start checking node reachability.
+		go util.Forever(func() {
+			if err = s.CheckNodeReachable(); err != nil {
```

Comment:
CheckNodeReachable is confusing.  It's not check node reachable, it's doing pod eviction.